doc/src/basics.md

```diff
@@ -30,20 +30,29 @@ info: downloading self-updates
 
 ## Keeping `rustup` up to date
 
-Running `rustup update` also checks for updates to `rustup` itself and automatically
-installs the latest version. To manually update `rustup` only,
-without updating installed toolchains, type `rustup self update`:
+## Keeping rustup up to date
+
+It is possible to control Rustup's automatic self update mechanism with the `auto-self-update` configuration
+variable.  This setting supports three values: `enable` and `disable` and `check-only`.
+
+* `disable` will ensure that no automatic self updating actions are taken.
+* `enable` will mean that `rustup update` and similar commands will also check-for, and install, any update to Rustup.
+* `check-only` will cause any automatic self update to check and report on any updates, but not to automatically install them.
+
+Whether automatic self-update is enabled or not, you can request that Rustup check for, and update it self to the
+latest version of `rustup` without updating any installed toolchains by running `rustup
+self update`:
 
 ```console
 $ rustup self update
 info: checking for self-updates
 info: downloading self-updates
 ```
 
-> #### Disable automatic self-updates
-> `rustup` will automatically update itself at the end of any toolchain installation.
-> You can prevent this automatic behaviour by passing the `--no-self-update` argument
-> when running `rustup update` or `rustup toolchain install`.
+**Note**: If auto-self-update is `enable`, `rustup` will automatically update itself at the end of any
+toolchain installation as well.  You can prevent this automatic behaviour by
+passing the `--no-self-update` argument when running `rustup update` or
+`rustup toolchain install`.
```

Comment:
A previous change deliberately called this out in a block quote with its own low-tier heading.  I suggest you don't undo that since it's a good readability improvement.

Reply:
Added.